pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-02-17 17:54
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def remove_create_subs(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    to_delete = subs.filter(type=10)
+    total = to_delete.count()
+    offset = 0
+    step = 10000
+    start = time.time()
+
+    while True:
+        units = to_delete[:step]
+        to_delete.filter(id__in=set(units.values_list("id", flat=True))).delete()
+
+        logger.debug(
+            "deleted %s/%s in %s seconds"
+            % (offset + step, total, (time.time() - start)))
+        if offset > total:
+            break
+        offset = offset + step
```

Comment:
I mean:

- we have 7 items to delete, offset is 0
- on first iteration:

  - it removes the 7 submissions,
  - 0 > 7 is False so doesn't break
  - then offset is incremented

- second iteration:

  - removes nothing
  - 10000 > 7 is True so aborts